Django 1.10 and 1.11 support

veryhappythings opened the pull request:
Hi, I saw in #12 that someone had attempted to add 1.11 support so I took the branch, polished it up, and added it to the test matrix. I think this works, but I'm a little concerned about the changes in utils.py - could someone who knows more about Django's URLs take a look and check that it makes sense please?

Thank you to @nisimpson for the original work found here: https://github.com/nisimpson/django-plugins/tree/django-1.11

veryhappythings committed:
Pin Django-dirtyfields to < 1.3

1.2 is the last build to support Django < 1.8

veryhappythings committed:
Enable support for 1.10

veryhappythings committed:
Remove deprecated NoArgsCommand

NoArgsCommand was deprecated in Django 1.10. BaseCommand takes
no arguments by default.

Thanks to Nathan Simpson (@nisimpson) for his work on this.

veryhappythings committed:
Removed deprecated patterns function

patterns was deprecated in 1.8 and removed in Django 1.10:
https://docs.djangoproject.com/en/2.1/releases/1.8/#django-conf-urls-patterns

Thanks to Nathan Simpson (@nisimpson) for his work on this.

veryhappythings committed:
Django 1.10 does not work with python 3.3

veryhappythings committed:
Test on Django 1.11

veryhappythings commented:
I've avoided losing compatibilty with older versions, but I've had to pin django-dirtyfields in order to do this. I think it'd be best to drop support in the next version, but that's up to you!

LilyFirefly commented on djangoplugins/management/commands/syncplugins.py:
```diff
@@ -3,26 +3,32 @@
-from django.core.management.base import NoArgsCommand
+
+try:
+    from django.core.management.base import BaseCommand
```
Can this import actually fail?

veryhappythings replied:
thinking about it, I don't think so, and worse, I don't think that the argument handling would work with an old version of Django here. I'll see if I can handle both versions reasonably.

veryhappythings committed:
Refactor syncplugins command to use BaseCommand

This should work for old and new versions of Django, relying on
slightly different APIs for BaseCommand instead of NoArgsCommand.

LilyFirefly commented on djangoplugins/management/commands/syncplugins.py:
```diff
@@ -18,6 +15,14 @@
+                        action='store_true',
+                        dest='delete',
+                        default=False,
+                        help='Delete poll instead of closing it'),
```
This help text doesn't match the version below.

veryhappythings committed:
Correct help text

veryhappythings commented:
Travis is passing - results are available on my fork https://travis-ci.org/veryhappythings/django-plugins

krischer commented:
Thanks a lot for this!

I don't do a lot of django development anymore and thus cannot devote a lot of time to this repository currently. Is anyone of you interested in taking over maintenance? I'd be happy to give you full access rights or even move the repository to one of your accounts or make a new group. Some with the pypi account. Let me know!

veryhappythings commented:
Hi, sorry for the delay in replying to this! Unfortunately we no longer use this library, so I don't think I'm the best equipped to take it over. Thank you for the offer, though!
